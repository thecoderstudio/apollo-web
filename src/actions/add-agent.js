const SELECT_OPERATING_SYSTEM = 'SELECT_OPERATING_SYSTEM';
const SELECT_ARCHITECTURE = 'SELECT_ARCHITECTURE';

function selectOperatingSystem(operatingSystem) {
  return {
    type: SELECT_OPERATING_SYSTEM,
    selectedOperatingSystem: operatingSystem
  };
}

function selectArchitecture(architecture) {
  return {
    type: SELECT_ARCHITECTURE,
    selectedArchitecture: architecture
  };
}

export {
  selectOperatingSystem,
  SELECT_OPERATING_SYSTEM,
  selectArchitecture,
  SELECT_ARCHITECTURE
};
