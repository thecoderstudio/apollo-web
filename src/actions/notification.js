let id = 0;
const severity = {
  INFO: 'info',
  WARNING: 'warning',
  ERROR: 'error',
  SUCCESS: 'success'
};

function notify(message, severity) {
  return {
    type: 'NOTIFY',
    id: id++,
    message,
    severity
  };
}

function dismiss(notificationId) {
  return {
    type: 'DISMISS',
    id: notificationId
  };
}

export { severity, notify, dismiss };
