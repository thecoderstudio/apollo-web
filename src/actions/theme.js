function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
