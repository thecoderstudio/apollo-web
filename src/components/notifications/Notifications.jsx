import React from 'react';
import { connect } from 'react-redux';
import { CSSTransitionGroup } from 'react-transition-group';
import styled from 'styled-components';
import Notification from './Notification';

const Container = styled(CSSTransitionGroup)`
  display: flex;
  flex-direction: column;
  align-items: center;
  position: absolute;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 2;
`;

const StyledNotification = styled(Notification)`
  margin 8px;

  &.fade-enter {
    opacity: 0.01;
  }

  &.fade-enter.fade-enter-active {
    opacity: 1;
    transition: opacity 200ms ease-in;
  }

  &.fade-leave {
    opacity: 1;
  }

  &.fade-leave.fade-leave-active {
    opacity: 0.01;
    transition: opacity 200ms ease-in;
  }
`;


function Notifications(props) {
  return (
    <Container
        transitionName="fade"
        transitionEnterTimeout={200}
        transitionLeaveTimeout={200}>
        {props.notifications.keySeq().map((id, index) => (
          <StyledNotification key={id} id={id} timeToLiveInSeconds={5} {...props.notifications.get(id)} />
        ))}
    </Container>
  );
}

export default connect(({ notifications }) => ({ notifications }))(Notifications);
