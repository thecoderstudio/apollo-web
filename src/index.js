import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
