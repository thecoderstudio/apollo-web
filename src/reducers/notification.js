import { Map } from 'immutable';

export default function notificationReducer(state = Map({}), action) {
  switch (action.type) {
    case 'NOTIFY':
      return state.set(action.id, {
        message: action.message,
        severity: action.severity
      });
    case 'DISMISS':
      return state.delete(action.id);
    default:
      return state;
  }
}
