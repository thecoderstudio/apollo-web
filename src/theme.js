export const darkTheme = {
  name: 'dark',
  black: '#181818',
  lightBlack: '#1B1B1B',
  InputBlack: '#1f1f1f',
  selectedBlack: '#252525',
  white: '#F8F8F8',
  inactive: '#363636',
  darkWhite: '#F1F1F1',
  darkGrey: '#3E3E3E',
  primary: '#02A3EE',
  accent: '#4D02EE',
  overlay: 'rgba(37, 37, 37, 0.6)',
  green: '#32E67E',
  red: '#ED3E2A',
  connectedColor: '#27AE60',
  disconnectedColor: '#C0392B',
  connectingColor: '#F39C12',
  success: '#27AE60',
  error: '#ED3E2A',
  warning: '#ED852A',
  boxShadow: '0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)',
  smallTextSize: '0.85rem'
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
