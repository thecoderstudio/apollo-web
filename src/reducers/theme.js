import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
