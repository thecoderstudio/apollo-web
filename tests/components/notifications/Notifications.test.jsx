import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { Map } from 'immutable';
import waitForExpect from 'wait-for-expect';
import { dismiss } from '../../../src/actions/notification';
import { darkTheme } from '../../../src/theme';
import Notifications from '../../../src/components/notifications/Notifications';

const mockStore = configureStore([]);

function getComponent(store) {
  return renderer.create(
    <Provider store={store}>
      <ThemeProvider theme={darkTheme}>
        <Notifications timeToLiveInSeconds={1} />
      </ThemeProvider>
    </Provider>
  );
}

describe('notifications', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      notifications: new Map({})
    });
  });

  it("renders correctly empty", () => {
    const tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders notifications and then dismisses them", async () => {
    store = mockStore({
      notifications: new Map({
        0: {
          message: "test 1",
          severity: 'info'
        },
        1: {
          message: "test 2",
          severity: 'warning'
        }
      })
    });
    store.dispatch = jest.fn();

    let tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();

    await waitForExpect(() => {
      expect(store.dispatch).toHaveBeenCalledTimes(2);
      tree = getComponent(store).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
