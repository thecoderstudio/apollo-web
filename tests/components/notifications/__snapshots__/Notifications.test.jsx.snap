// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notifications renders correctly empty 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  left: 0px;
  right: 0px;
  bottom: 32px;
  z-index: 2;
}

<span
  className="c0"
/>
`;

exports[`notifications renders notifications and then dismisses them 1`] = `
.c1 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #4D02EE;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c5 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #ED852A;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c6 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #27AE60;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c3 {
  float: right;
  color: white;
}

.c4 {
  color: white;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  left: 0px;
  right: 0px;
  bottom: 32px;
  z-index: 2;
}

.c2 {
  margin: 8px;
}

.c2.fade-enter {
  opacity: 0.01;
}

.c2.fade-enter.fade-enter-active {
  opacity: 1;
  -webkit-transition: opacity 200ms ease-in;
  transition: opacity 200ms ease-in;
}

.c2.fade-leave {
  opacity: 1;
}

.c2.fade-leave.fade-leave-active {
  opacity: 0.01;
  -webkit-transition: opacity 200ms ease-in;
  transition: opacity 200ms ease-in;
}

<span
  className="c0"
>
  <div
    className="c1 c2"
    color="#4D02EE"
  >
    <i
      className="c3 fas fa-times"
      onClick={[Function]}
    />
    <h5
      className="c4"
    >
      test 1
    </h5>
  </div>
  <div
    className="c5 c2"
    color="#ED852A"
  >
    <i
      className="c3 fas fa-times"
      onClick={[Function]}
    />
    <h5
      className="c4"
    >
      test 2
    </h5>
  </div>
  <div
    className="c6 c2"
    color="#27AE60"
  >
    <i
      className="c3 fas fa-times"
      onClick={[Function]}
    />
    <h5
      className="c4"
    >
      test 3
    </h5>
  </div>
</span>
`;

exports[`notifications renders notifications and then dismisses them 2`] = `
.c1 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #4D02EE;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c5 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #ED852A;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c6 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #27AE60;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c3 {
  float: right;
  color: white;
}

.c4 {
  color: white;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  left: 0px;
  right: 0px;
  bottom: 32px;
  z-index: 2;
}

.c2 {
  margin: 8px;
}

.c2.fade-enter {
  opacity: 0.01;
}

.c2.fade-enter.fade-enter-active {
  opacity: 1;
  -webkit-transition: opacity 200ms ease-in;
  transition: opacity 200ms ease-in;
}

.c2.fade-leave {
  opacity: 1;
}

.c2.fade-leave.fade-leave-active {
  opacity: 0.01;
  -webkit-transition: opacity 200ms ease-in;
  transition: opacity 200ms ease-in;
}

<span
  className="c0"
>
  <div
    className="c1 c2"
    color="#4D02EE"
  >
    <i
      className="c3 fas fa-times"
      onClick={[Function]}
    />
    <h5
      className="c4"
    >
      test 1
    </h5>
  </div>
  <div
    className="c5 c2"
    color="#ED852A"
  >
    <i
      className="c3 fas fa-times"
      onClick={[Function]}
    />
    <h5
      className="c4"
    >
      test 2
    </h5>
  </div>
  <div
    className="c6 c2"
    color="#27AE60"
  >
    <i
      className="c3 fas fa-times"
      onClick={[Function]}
    />
    <h5
      className="c4"
    >
      test 3
    </h5>
  </div>
</span>
`;
